snuba/consumers/consumer_builder.py

```diff
@@ -280,6 +282,56 @@ def build_streaming_strategy_factory(
 
         return strategy_factory
 
+    def build_dlq_strategy_factory(
+        self, instruction: DlqInstruction
+    ) -> ProcessingStrategyFactory[KafkaPayload]:
+        """
+        Similar to build_streaming_strategy_factory with 2 differences:
+        - uses ExitAfterNMessages instead of the standard CommitOffsets strategy
+        - never commits to the commit log (the commit log should always be monotonic)
+        """
+        table_writer = self.storage.get_table_writer()
+        stream_loader = table_writer.get_stream_loader()
+
+        logical_topic = stream_loader.get_default_topic_spec().topic
+
+        processor = stream_loader.get_processor()
+
+        def build_commit_strategy(commit: Commit) -> ProcessingStrategy[Any]:
```

Comment:
I think it would make more sense to add `max_messages_to_process: Optional[int]` as a parameter to `KafkaConsumerStrategyFactory` -- since for other strategies like multiprocessing we also forward args

Reply:
yeah you're right, will update